packages/react-native/lib/credentials/credentialHooks.js

```diff
@@ -35,8 +36,25 @@ export function getCredentialTimestamp(credential) {
   return new Date(credential.issuanceDate).getTime() || 0;
 }
 
+export function waitFor(condition, timeout) {
+  return new Promise((resolve, reject) => {
+    const interval = setInterval(async () => {
+      if (await Promise.resolve(condition())) {
+        clearInterval(interval);
+        resolve(true);
+      }
+    }, 400);
+
+    setTimeout(() => {
+      clearInterval(interval);
+      reject(new Error('Timed out'));
+    }, timeout);
+  });
+}
+
 async function getCredentialValidityStatus(credential) {
   try {
+    await waitFor(() => dockService.isApiConnected(), 8000);
```

Comment:
8 seconds seems like a fairly long time out - are we hitting that regularly?

Reply:
No, can take 3 to 4 seconds. But a bad internet connection could reach almost 8 seconds.